app/controllers/listings_controller.rb

```diff
@@ -1,5 +1,14 @@
 class ListingsController < ApplicationController
   def index
-    @listings = Listing.all
+    @listings = Listing.where(nil)
+    filtering_params(params).each do |key, value|
+      @listings = @listings.public_send(key, value) if value.present?
```

Comment:
if you search for a city, then the key is city and the value is your "search city."  Public send then uses the key as a public method call on listings to search for your "search city.